### PR TITLE
refine bbox_normalize in infer.py

### PR DESCRIPTION
**Pull request by heavengate**
**Refine is_bbox_normalized check in PaddleDetection infer.py**
1. add is_bbox_normalized() in SSD
2. tools/infer.py get is_bbox_normalized from model

### CODE REVIEW

**Comment by heavengate** [comment]
since is_bbox_normalized is determined by output layer and is not configurable, make it as a attribute of models, and only SSD is_bbox_normalized = True.
please help to review~ @qingqing01 @willthefrog

File: PaddleCV/PaddleDetection/ppdet/modeling/architectures/base.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from __future__ import division
+from __future__ import print_function
+
+
+class ArchitectureBase(object):
+    """
+    Base object for architectures.
+    """
+
+    __category__ = 'architecture'
```
Comment (willthefrog) [inline PaddleCV/PaddleDetection/ppdet/modeling/architectures/base.py]: not needed as it is not registered

**Comment by willthefrog** [comment]
IMHO, a base class seems a bit overkill for just this purpose, adding a attribute (or class attribute) and checking with `hasattr` should suffice.

Currently the model  interfaces are defined by convention, which is more pythonic. Interface by inheritance (a la java) does not add much as most of the methods are one-liners any way.

In my initial design there is only a `build(feed_var, mode)` method, all these `train`/`test` /`eval` are just shortcuts for backward compatibility during the refactor. Come to think about it now, these shortcuts should be simply removed.

**Comment by heavengate** [comment]
> IMHO, a base class seems a bit overkill for just this purpose, adding a attribute (or class attribute) and checking with `hasattr` should suffice.
> 
> Currently the model interfaces are defined by convention, which is more pythonic. Interface by inheritance (a la java) does not add much as most of the methods are one-liners any way.
> 
> In my initial design there is only a `build(feed_var, mode)` method, all these `train`/`test` /`eval` are just shortcuts for backward compatibility during the refactor. Come to think about it now, these shortcuts should be simply removed.

Thanks, done, please help to review~

File: PaddleCV/PaddleDetection/tools/infer.py [hunk 2]
```diff
@@ -164,6 +164,11 @@ def main():
     clsid2catid, catid2name = get_category_info(anno_file, with_background,
                                                 use_default_label)
 
+    # whether output bbox is normalized in model output layer
+    is_bbox_normalized = False
+    if hasattr(model, 'is_bbox_normalized'):
```
Comment (willthefrog) [inline PaddleCV/PaddleDetection/tools/infer.py]: check callable? or simply use a member variable instead of method?

Comment (heavengate) [reply PaddleCV/PaddleDetection/tools/infer.py]: Done, thanks

**Review by willthefrog — APPROVED** [review]
LGTM

**Pull request merged**